src/XMakeTasks/ManifestUtil/SecurityUtil.cs

```diff
@@ -762,20 +762,33 @@ internal static string GetPathToTool(System.Resources.ResourceManager resources)
         {
 #pragma warning disable 618 // Disabling warning on using internal ToolLocationHelper API. At some point we should migrate this.
             string toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.VersionLatest, VisualStudioVersion.VersionLatest);
-            if (toolPath == null)
-                toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.Version45, VisualStudioVersion.Version110);
-            if (toolPath == null)
+            if (toolPath == null || !File.Exists(toolPath))
+            {
+                toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.Version45,
+                    VisualStudioVersion.Version110);
+            }
+            if (toolPath == null || !File.Exists(toolPath))
             {
                 var pathToDotNetFrameworkSdk = ToolLocationHelper.GetPathToDotNetFrameworkSdk(TargetDotNetFrameworkVersion.Version40, VisualStudioVersion.Version100);
                 if (pathToDotNetFrameworkSdk != null)
                 {
                     toolPath = Path.Combine(pathToDotNetFrameworkSdk, "bin", ToolName);
                 }
             }
-            if (toolPath == null)
+            if (toolPath == null || !File.Exists(toolPath))
+            {
+                toolPath = GetVersionIndependentToolPath(ToolName);
```

Comment:
I would say it makes sense to have it last I would say, not first as we want this to be only of last resort when there is no SDK installed. I left the check for the users current directory after this one as the error message will point to the file missing from his project directory and it would make more sense to go online, download it and put it there instead of some "random" folder from his perspective